core/src/main/java/io/fabric8/maven/core/openshift/BuildService.java

```diff
@@ -81,6 +81,15 @@ public void waitForOpenShiftBuildToComplete(OpenShiftClient client, Build build)
                                                   "Failed to tail build log", logTerminateLatch, log);
             Watcher<Build> buildWatcher = getBuildWatcher(latch, buildName, buildHolder);
             try (Watch watcher = client.builds().withName(buildName).watch(buildWatcher)) {
+                // Check if the build is already finished to avoid waiting indefinitely
+                Build lastBuild = client.builds().withName(buildName).get();
+                String lastStatus = KubernetesResourceUtil.getBuildStatusPhase(lastBuild);
+                if (Builds.isFinished(lastStatus)) {
+                    log.debug("Build %s is already finished", buildName);
+                    buildHolder.set(lastBuild);
+                    latch.countDown();
+                }
+
```

Comment:
Couldn't this block just be moved up to the beginning of the method and just return if the build already has finished ? would make the logic a bit simpler. 

Or is it too early ? But I guess when this method is called in 3.7 the build is already finished ?

Also I wonder whether this is really true all the case as a build can take quite a long time and I wonder how this is dealt with timeouts on the API. I know that openshift used to uses "Expect: 100-continue" 
 HTTP header and this is supported in the latest OkHttp client. I opened an PR ehttps://github.com/fabric8io/kubernetes-client/pull/620 for this but I didn't really follow this. It was for timeouts happening in an error case. Not sure whether this is related to the issue here.